azext_iot/central/providers/device_provider.py

```diff
@@ -181,8 +181,11 @@ def get_device_registration_info(
         if info:
             return info
 
-        device = self.get_device(device_id)
-        if device.device_status == DeviceStatus.provisioned:
+        device = self.get_device(device_id, central_dns_suffix)
+        if (
+            DeviceStatus(device.get_registration_info().get("device_status"))
```

Comment:
I think this can be reduced to `if device.device_status == DeviceStatus.provisioned`